### DISCUSSION TITLE
Peacock crashing

### DISCUSSION THREAD

**Discussion by g1202030203** [idaholab/moose#17357] [Q&A Getting Started] [answered]
Hi All,

I tried to run the peacock by run "./peacock", But it always shows an error like this "zsh: segmentation fault  ./peacock" and python quit unexpectedly like the following:  

process:               python3.7 [24078]
Path:                  /Users/USER/*/python
Identifier:            python3.7
Version:               0
Code Type:             X86-64 (Native)
Parent Process:        zsh [23344]
Responsible:           Terminal [432]
User ID:               501

How to solve this problem?

**Comment by GiudGiud** [comment, marked as answer]
Hi,

You seem to be using the system python. Did you install moose using conda? If so you need to use the conda python. 

Guillaume

**Reply by g1202030203** [reply]
Hi,

I think I installed moose using conda. I followed the installation instructions from the moose website: https://mooseframework.inl.gov/getting_started/installation/conda.html

Ethan

**Reply by GiudGiud** [reply]
what does `which python` return?
We need peacock to use the python from conda

**Reply by g1202030203** [reply]
It returns to "/Users/chongjie/miniconda3/envs/moose/bin/python"

**Reply by GiudGiud** [reply]
Ok it might be a different issue then
@aeslaughter will know more about Peacock failures on mac

**Reply by aeslaughter** [reply]
Are you using BigSur? It was only recently that we have peacock working with it. @milljm did you get BigSur working?

**Reply by g1202030203** [reply]
Yes, I use BigSur.

**Reply by milljm** [reply]
Unfortunately, Big Sur won't work, until this PR makes it in: #17106, #17105

A work-around, would be for you to build these Conda packages yourself. If you're willing, I can walk you through it. It will require sudo rights.

**Reply by ykvishal** [reply]
Hi @milljm ,
Can you explain the necessary steps for this issue? I am having the same issue with Peacock on Bigsur.

Thank you.

**Reply by milljm** [reply]
I had a similar question earlier. I'll paste here what I wrote them:

```bash
cd moose
# add my remote (for the updated recipes necessary for BigSur).
git remote add jason https://github.com/milljm/moose
git fetch jason
git checkout jason/17105-python3.9
# get out of the moose environment if it's activated
conda deactivate
cond install conda-build
cd conda/mpich
conda build .
cd ../libmesh_vtk
conda build . --use-local
cd ../petsc
conda build . --use-local
cd ../libmesh
conda build . --use-local
cd ../tools
conda build .
```

If all goes well, you'll have all the moose-packages built locally. To install these into a new environment:
```bash
conda create -n bigsur --use-local moose-tools
conda activate bigsur
conda install --use-local moose-libmesh
```
Hopefully that works! Once installed, you can safely move back to using moose/master head.

**Reply by ykvishal** [reply]
Thank you. 

I am getting error after following step

> cd conda/mpich

> conda build .

Last few lines before this error

> checking build system type... x86_64-apple-darwin13.4.0
> checking host system type... x86_64-apple-darwin13.4.0
> checking target system type... x86_64-apple-darwin13.4.0
> checking for x86_64-apple-darwin13.4.0-icc... x86_64-apple-darwin13.4.0-clang
> checking whether the C compiler works... no
> configure: error: in `/Users/v.yadav/miniconda3/conda-bld/moose-mpich_1623175426532/work':
> configure: error: C compiler cannot create executables
> See `config.log' for more details
> Traceback (most recent call last):
>   File "/Users/v.yadav/miniconda3/bin/conda-build", line 11, in <module>
>     sys.exit(main())
>   File "/Users/v.yadav/miniconda3/lib/python3.8/site-packages/conda_build/cli/main_build.py", line 481, in main
>     execute(sys.argv[1:])
>   File "/Users/v.yadav/miniconda3/lib/python3.8/site-packages/conda_build/cli/main_build.py", line 470, in execute
>     outputs = api.build(args.recipe, post=args.post, test_run_post=args.test_run_post,
>   File "/Users/v.yadav/miniconda3/lib/python3.8/site-packages/conda_build/api.py", line 186, in build
>     return build_tree(
>   File "/Users/v.yadav/miniconda3/lib/python3.8/site-packages/conda_build/build.py", line 3068, in build_tree
>     packages_from_this = build(metadata, stats,
>   File "/Users/v.yadav/miniconda3/lib/python3.8/site-packages/conda_build/build.py", line 2191, in build
>     utils.check_call_env(cmd, env=env, rewrite_stdout_env=rewrite_env,
>   File "/Users/v.yadav/miniconda3/lib/python3.8/site-packages/conda_build/utils.py", line 412, in check_call_env
>     return _func_defaulting_env_to_os_environ('call', *popenargs, **kwargs)
>   File "/Users/v.yadav/miniconda3/lib/python3.8/site-packages/conda_build/utils.py", line 392, in _func_defaulting_env_to_os_environ
>     raise subprocess.CalledProcessError(proc.returncode, _args)
> subprocess.CalledProcessError: Command '['/bin/bash', '-o', 'errexit', '/Users/v.yadav/miniconda3/conda-bld/moose-mpich_1623175426532/work/conda_build.sh']' returned non-zero exit status 77.

**Reply by milljm** [reply]
You'll need the Xcode SDK 10.9 installed at /opt/. I like to grab the SDK from here: https://github.com/phracker/MacOSX-SDKs.
